### ISSUE TITLE
3080 ti(GPU 0) and 3070 ti(GPU 1), however only GPU 0 is working while GPU 1 always gets an error.

### ISSUE THREAD

**Issue by LaVorz** [rbbrdckybk/dream-factory#2]
Tried setting the working GPU to GPU 1, but still getting the same error. could not get a simultaneous generation going.

This is the error:

RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:1 and cuda:0! (when checking argument for argument index in method wrapper__index_select)

this is what I've found online about it, but I do not know where to begin to troubleshoot the issue
https://discuss.pytorch.org/t/runtimeerror-expected-all-tensors-to-be-on-the-same-device-but-found-at-least-two-devices-cuda-1-and-cuda-0-when-checking-argument-for-argument-index-in-method-wrapper-index-select-while-using-dataparallel-class/143566/8

**Comment by rbbrdckybk** [comment]
Can you please copy & paste both your config.txt file and the prompt file that you're trying to run? Also what OS are you on? Thanks!

**Comment by Belial52** [comment]
Since this appears to be abandoned and I'm getting the same issue at the moment, here's some information. System is Windows 11. Trying to do this on a pair of matched 1080tis. I notice that this is the error I get when I don't run in low memory mode, but when I run this in low memory mode, the second GPU just hangs and does nothing.

Error:
`Traceback (most recent call last):                                                              | 0/50 [00:00<?, ?it/s]
  File "H:\Stable_Diffusion\dream-factory\stable-diffusion\scripts_mod\txt2img.py", line 387, in <module>
    main()
  File "H:\Stable_Diffusion\dream-factory\stable-diffusion\scripts_mod\txt2img.py", line 319, in main
    uc = model.get_learned_conditioning(batch_size * [""])
  File "H:\Stable_Diffusion\dream-factory\stable-diffusion\ldm\models\diffusion\ddpm.py", line 554, in get_learned_conditioning
    c = self.cond_stage_model.encode(c)
  File "H:\Stable_Diffusion\dream-factory\stable-diffusion\ldm\modules\encoders\modules.py", line 162, in encode
    return self(text)
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\torch\nn\modules\module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "H:\Stable_Diffusion\dream-factory\stable-diffusion\ldm\modules\encoders\modules.py", line 156, in forward
    outputs = self.transformer(input_ids=tokens)
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\torch\nn\modules\module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\transformers\models\clip\modeling_clip.py", line 733, in forward
    return self.text_model(
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\torch\nn\modules\module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\transformers\models\clip\modeling_clip.py", line 636, in forward
    hidden_states = self.embeddings(input_ids=input_ids, position_ids=position_ids)
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\torch\nn\modules\module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\transformers\models\clip\modeling_clip.py", line 165, in forward
    inputs_embeds = self.token_embedding(input_ids)
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\torch\nn\modules\module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\torch\nn\modules\sparse.py", line 160, in forward
    return F.embedding(
  File "C:\Users\\anaconda3\envs\dream-factory\lib\site-packages\torch\nn\functional.py", line 2210, in embedding
    return torch.embedding(weight, input, padding_idx, scale_grad_by_freq, sparse)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:1 and cuda:0! (when checking argument for argument index in method wrapper__index_select)`

Prompt:
`[config]

!MODE = standard
!SD_LOW_MEMORY = no	
!SD_LOW_MEM_TURBO = ye
!DELIM = ","
!REPEAT = no

!WIDTH = 512							
!HEIGHT = 512							
!STEPS = 50							
!SAMPLER = ddim                      
!SAMPLES = 2					      	
!SCALE = 7.5							
!INPUT_IMAGE = 						
!STRENGTH = 0.75						
!CKPT_FILE = models/ldm/stable-diffusion-v1/Anything-V3.0.ckpt	

!USE_UPSCALE = no						
!UPSCALE_AMOUNT = 2.0					
!UPSCALE_FACE_ENH = no				
!UPSCALE_KEEP_ORG = no				

!NEG_PROMPT = ugly,morbid,extra fingers,poorly drawn hands,mutation,blurry,extra limbs,gross proportions,missing arms,mutated hands,long neck,duplicate,mutilated,mutilated hands,poorly drawn face,deformed,bad anatomy,cloned face,malformed limbs,missing legs,too many fingers

[prompts]
cat ears, cat tail, studio ghibli style, cute, blonde hair, blue eyes, brown highlights
cat ears, cat tail, studio ghibli style, cute, blonde hair, blue eyes, brown highlights`

Config:

`USE_GPU_DEVICES = 0,1

PROMPTS_LOCATION = prompts

OUTPUT_LOCATION = output

WEBSERVER_USE = yes

WEBSERVER_PORT = 8080

WEBSERVER_NETWORK_ACCESSIBLE = no

WEBSERVER_USE_AUTHENTICATION = no

WEBSERVER_AUTH_USERNAME = admin
WEBSERVER_AUTH_PASSWORD = password

WEBSERVER_OPEN_BROWSER = yes

WEBSERVER_CONSOLE_LOG = no

GALLERY_MAX_IMAGES = 200

GALLERY_REFRESH = 30

GALLERY_USER_FOLDER = 
GALLERY_USER_FOLDER_ALIAS = favorites


PF_SD_LOW_MEMORY = no	
PF_SD_LOW_MEM_TURBO = yes
PF_WIDTH = 512
PF_HEIGHT = 512  
PF_SAMPLER = euler_a
PF_STEPS = 50        
PF_SCALE = 7.5    
PF_SAMPLES = 1	
PF_USE_UPSCALE = no
PF_UPSCALE_AMOUNT = 2.0	
PF_UPSCALE_FACE_ENH = no	
PF_UPSCALE_KEEP_ORG = no
PF_CKPT_FILE =`

**Comment by rbbrdckybk** [comment]
Thanks for the info dump - everything looks fine in your config and prompt files to me. 

Unfortunately I took apart my multi-GPU machine a couple weeks ago and currently only have access to single-GPU computers, so I don't have an easy way to troubleshoot on my end. The errors you're seeing are from the stable diffusion side (e.g. not the Dream Factory code itself), and it's possible that something got messed up recently when support for the newer samplers was added. 

I'm currently looking into picking one of the better-supported repos and using that as the SD backend vs trying to slowly add features myself to the base SD and basujindal's fork, which seems to be mostly abandoned. Probably won't be able to accomplish a ton until after the holidays though, so unfortunately you'll probably have to stick to a single GPU until I can figure things out.
